firmware/fw_base.S

```diff
@@ -844,7 +844,7 @@ _clear_prev:
 	ret
 
 #ifdef FW_FDT_PATH
-	.section .fdt
+	.section .fdt, "a", %progbits
```

Comment:
@collinmay Do you have a link to an appropriate reference that explains the syntax here, or could you explain? My understanding is the .fdt section is being marked as "allocatable", which I think means the section is loaded to memory at runtime. Based on this manual (https://sourceware.org/binutils/docs/as/Section.html), it seems like the progbits should be `@progbits` though, not `%progbits`? And that type argument just says the section contains data.

Reply:
Yeah, that's the reference. The allocatable flag is needed to ensure the section gets covered by a program header and loaded into memory at runtime. Otherwise the linker seems to be allowed to skip covering it with a program header at its own discretion, which is wholly unhelpful.

I didn't notice the difference in punctuation, since I saw the rest of the file using `%progbits` and copied that. I don't think the difference in punctuation is important.
> Note on targets where the @ character is the start of a comment (eg ARM) then another character is used instead. For example the ARM port uses the % character. 

See also https://github.com/bminor/binutils-gdb/blob/d1a24139adfb2f029c9d52a9c43da44a617f90a4/gas/config/obj-elf.c#L1221

I'm not really sure what progbits means beyond what's unhelpfully written in the manual, but it does indeed contain data and I think it helps make the linker less likely to discard the section outright. Linkers have so much weird, underdocumented special-casey behavior...

Reply:
Thanks! I'm trying to understand all of the code we've added/modified relative to the upstream repo. It does look like our changes rebase cleanly onto the latest upstream. However, the way we have the FDT section setup is that it exists after the payload in the binary, and if the binary is greater than 32 MB it will clobber the FDT and fail (e.g., if our Linux image is > 32 MB, which is very easy to do when doing an FS overlay with benchmark programs and large input files)

Reply:
I don't understand any of the subtleties but the closer we can get to upstream the better :)